sviatoslav/src/main/java/com/cbu/edu/hw_01/ImitationStack.java

```diff
@@ -0,0 +1,89 @@
+package com.cbu.edu.hw_01;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Created by Sviatoslav on 20.02.2017.
+ */
+public class ImitationStack<T extends Number> implements IGetIterator{
+    private int size;
+    private T[] stack;
+    private T[] newStack;
+    private int top;
+
+
+
+    public ImitationStack(int size) {
+        this.size = size;
+        stack = (T[])new Object[size];
+        this.top = -1;
+    }
+
+    public boolean isEmpty() {
+        boolean check = false;
+        if (top == -1) {
+            check = true;
+        }
+        return check;
+    }
+
+    public void push(T element) {
+        if (stack.length < size) {
+            int top = ++this.top;
+            stack[top] = element;
+        } else if (stack.length >= size) {
+            newStack = Arrays.copyOf(stack, size * 3 / 2 + 1);
+            stack = null;
+            stack = newStack;
+            int top = ++this.top;
+            stack[top] = element;
+        }
```

Comment:
You have code duplication at 34 and 40 lines:
```
int top = ++this.top;
stack[top] = element;
```
It may be possible to rewrite somehow this design without duplicating?